src/Controller/CommandLineController.ts

```diff
@@ -0,0 +1,47 @@
+import yargs from 'yargs'
+
+export interface JunknetArguments {
+	makefile: string
+	docker_image: string
+	target: string | null
+}
+
+/**
+ * Interpret a list of arguments, returning the relevant ones in the f
+ *
+ * @param argv - the invoking process's command-line arguments.
+ * @returns JunknetInterface object containing option values.
+ */
+export function interpretArgv(argv: readonly string[]): JunknetArguments {
+	// Use only the first two elements; Node.js appends extra elements to process.argv.
+	const yargsArgv = yargs(argv.slice(2)).options({
+		f: {
+			alias: 'makefile',
+			type: 'string',
+			default: 'Makefile',
+			desc: 'The Makefile to process',
+		},
+		i: {
+			alias: 'docker-image',
+			type: 'string',
+			default: 'ubuntu:18.04',
```

Comment:
Don't provide an arbitrary default.

Btw, this image doesn't contain `make`.

Reply:
Sure. I'll make the dockerfile the first positional argument.